tests/rptest/services/redpanda.py

```diff
@@ -117,6 +117,47 @@ def one_or_many(value):
         return value
 
 
+class ResourceSettings:
+    """
+    Control CPU+memory footprint of Redpanda instances.  Pass one
+    of these into your RedpandaTest constructor if you want to e.g.
+    create low-memory situations.
+    """
+    def __init__(self,
+                 num_cpus=None,
```

Comment:
any objection to using `*` to separate kwargs from position arguments (ie `...(self, *, num_cpus=...)`)? using Pyright I found that it was really helpful when refactoring function parameters because with `*` it can't know that when the last positional argument is deleted that it isn't supposed to be the first kwarg.